sql/binlog.cc

```diff
@@ -9982,6 +10151,15 @@ int MYSQL_BIN_LOG::ordered_commit(THD *thd, bool all, bool skip_commit)
     else if (check_purge)
       purge();
   }
+#ifdef HAVE_REPLICATION
+  else
```

Comment:
Why is this in "else"? Shouldn't purging by size happen regardless of the if condition?

Reply:
I used `else` because in case of `true` there is `purge()` which also unconditionally calls `purge_logs_by_size(true)`. But you are right that without `else` it should be simpler and more readable.